### DISCUSSION TITLE
Vue + SSR streaming + Suspense + partial/progressive hydration

### DISCUSSION THREAD

**Discussion by lourot** [vikejs/vike#1318] [Discussion]
Spin-off of #1271. I seem to not be able to get SSR streaming + partial hydration working with Vue 3. See https://github.com/vikejs/vike/pull/1271#discussion_r1406044526 for more details. Investigating

**Comment by nhattran998** [comment]
Hope this feature will be add soon 😆

**Reply by nhattran998** [reply]
Or do you have some hint? Might be I can work on it!

**Reply by lourot** [reply]
@harrytran998 I'd love to get some help exploring this! I ended up working on other things meanwhile.

Here is a reproducer: https://github.com/vikejs/vike/tree/aurelien/reproducer/discussion-1318/test/stream-vue-onServerPrefetch

I have no clue what is wrong. The problem could be in Vike, in Vue, or in the example itself.  So my next step would be to create a simpler reproducer *without Vike*.

**Comment by lourot** [comment]
@phonzammi has created a reproducer without Vike :tada: https://github.com/phonzammi/vite-vue-streaming-ssr

Also he has reached out to `create-vite-extra` [here](https://github.com/bluwy/create-vite-extra/issues/45) to validate the design :+1:

This is so useful, thank you again @phonzammi! Let's create a Vue discussion to ask them about progressive hydration.

**Reply by brillout** [reply]
@phonzammi Neat 💯 We actually always wanted to get around that. Very much looking forward to hear what the Vue folks are going to say 👀 (also feel free to present yourself as a Vike team member).

**Reply by phonzammi** [reply]
Thanks for you both too. Let @AurelienLourot & me prepare for the text first.

**Reply by lourot** [reply]
In the end @phonzammi found evidence that Vue knows already that they don't support this, so this is something Vike also can't support at this point. Thanks a lot @phonzammi for the great research :pray: 

### Details

In a nutshell, for progressive hydration to work, at least two Vue features would be necessary:

1. the ability to render Suspense fallbacks on server, instead of hanging and rendering Suspense defaults.
2. the ability to then stream some `<script>document.getElementById(...` at the end of the body. (See https://github.com/phonzammi/vite-vue-streaming-ssr/pull/1 for more details)

Here is evidence that Vue knows they don't support 1. yet:
* https://github.com/vuejs/core/pull/8555 (not merged yet at time of writing)
* https://github.com/vuejs/core/blob/bb6b7a297e51f8b9999922a079fb4ceef000758a/packages/runtime-core/src/components/Suspense.ts#L793-L798 (at time of writing, SSR can render Suspense success or failure but not fallback)

**Reply by phonzammi** [reply]
Thanks @AurelienLourot for the summary,
I can't put the best words like you did 👍
